src/Fmt_ast.ml

```diff
@@ -1978,25 +1978,41 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                $ fmt_expression c (sub_exp ~ctx e2) )
            $ fmt_atrs ))
   | Pexp_sequence _ ->
-      let is_simple x = is_simple c.conf width x in
-      let break xexp1 xexp2 = not (is_simple xexp1 && is_simple xexp2) in
+      let break xexp1 xexp2 =
+        match c.conf.break_sequences with
+        | `Preserve ->
+            let l1 = xexp1.ast.pexp_loc.loc_end.pos_lnum in
+            let l2 = xexp2.ast.pexp_loc.loc_start.pos_lnum in
+            l2 - l1 > 1
+        | `Fit | `All -> false
+      in
       let grps = List.group (Sugar.sequence c.cmts xexp) ~break in
       let fmt_grp grp =
-        list grp
-          ( match c.conf.sequence_style with
-          | `Separator when c.conf.break_sequences -> " ;@;<1000 0>"
-          | `Separator -> " ;@ "
-          | `Terminator when c.conf.break_sequences -> ";@;<1000 0>"
-          | `Terminator -> ";@ " )
-          (fmt_expression c)
+        list_pn grp (fun ?prev x ?next:_ ->
+            ( match prev with
+            | Some prev ->
+                let is_simple x = is_simple c.conf width x in
+                let break = not (is_simple prev && is_simple x) in
+                fmt
```

Comment:
Minor comment: You could remove an indentation level by calling `fmt` every time.

Reply:
fixed